portal/get_patient_info.php

```diff
@@ -63,55 +63,78 @@
 // Authentication
 require_once('../interface/globals.php');
 require_once(dirname(__FILE__) . "/lib/appsql.class.php");
-$logit = new ApplicationTable();
 require_once("$srcdir/authentication/common_operations.php");
 require_once("$srcdir/user.inc");
-$password_update = isset($_SESSION['password_update']);
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+
+$logit = new ApplicationTable();
+$password_update = isset($_SESSION['password_update']) ? $_SESSION['password_update'] : 0;
 unset($_SESSION['password_update']);
-$plain_code = $_POST['pass'];
+$lookup = $_POST['uname'];
+if ($password_update === 2 && !empty($_SESSION['forward'])) {
+    // a one time auth token
+    $lookup = $_SESSION['forward'];
+    unset($_SESSION['forward']);
+}
 
 $authorizedPortal = false; // flag
 DEFINE("TBL_PAT_ACC_ON", "patient_access_onsite");
 DEFINE("COL_PID", "pid");
 DEFINE("COL_POR_PWD", "portal_pwd");
 DEFINE("COL_POR_USER", "portal_username");
+DEFINE("COL_POR_LOGINUSER", "portal_login_username");
 DEFINE("COL_POR_SALT", "portal_salt");
 DEFINE("COL_POR_PWD_STAT", "portal_pwd_status");
+
+
+// This is now an account id. Legacy username. Will force case sensitive regardless of collation
 $sql = "SELECT " . implode(",", array(
-    COL_ID,
-    COL_PID,
-    COL_POR_PWD,
-    COL_POR_SALT,
-    COL_POR_PWD_STAT
-)) . " FROM " . TBL_PAT_ACC_ON . " WHERE " . COL_POR_USER . "=?";
-$auth = privQuery($sql, array(
-    $_POST['uname']
-));
+        COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+    " WHERE BINARY " . COL_POR_LOGINUSER . "= ?";
+$auth = privQuery($sql, array($lookup));
+
+if ($password_update === 2) {
+    $validate = substr($auth[COL_POR_LOGINUSER], 32, 6);
+    $move_on = ($validate === $_SESSION['pin']) === ($_POST['token_pin'] === $validate);
```

Comment:
quick logic question here:
if both above comparisons are false, won't it then make $move_on true?

Reply:
true, really only need one check but i'm trying to be too clever. plus, I always think things will turn out positive:)